postgres/postgres.go

```diff
@@ -0,0 +1,207 @@
+package postgres
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/ONSdigital/dp-import-api/models"
+	"github.com/ONSdigital/dp-import-api/utils"
+	"strings"
+	"time"
+)
+
+// Datastore - A structure to hold SQL statements to be used to gather information or insert about Jobs and instances
+type Datastore struct {
+	db                   *sql.DB
+	addJob               *sql.Stmt
+	updateState          *sql.Stmt
+	addInstance          *sql.Stmt
+	findInstance         *sql.Stmt
+	addS3Url             *sql.Stmt
+	addEvent             *sql.Stmt
+	addDimension         *sql.Stmt
+	getDimensions        *sql.Stmt
+	addNodeID            *sql.Stmt
+	createPublishMessage *sql.Stmt
+}
+
+func prepare(sql string, db *sql.DB) *sql.Stmt {
+	statement, err := db.Prepare(sql)
+	if err != nil {
+		panic(err)
+	}
+	return statement
+}
+
+// NewDatastore - Create a postgres datastore. This is used to store and find information about jobs and instances.
+func NewDatastore(db *sql.DB) (Datastore, error) {
+	addJob := prepare("INSERT INTO Jobs(job) VALUES($1) RETURNING jobId", db)
+	updateState := prepare("UPDATE Jobs SET job = jsonb_set(job, '{state}', TO_JSONB($1::TEXT)) WHERE jobId = $2 RETURNING jobId", db)
+	addInstance := prepare("INSERT INTO Instances(jobId, instance) VALUES($1, $2) RETURNING instanceId", db)
+	findInstance := prepare("SELECT instance FROM Instances WHERE instanceId = $1", db)
+	addS3Url := prepare("UPDATE Jobs SET job = jsonb_set(job, '{files}', (SELECT (job->'files')  || TO_JSONB(json_build_object('alaisName',$1::TEXT,'url',$2::TEXT)) FROM Jobs WHERE jobId = $3), true) WHERE jobId = $3 RETURNING jobId", db)
```

Comment:
is this just appending a URL onto an existing list? I am unfamiliar with the query syntax, but just double check its not overwriting any existing entries.